core/src/test/java/org/apache/calcite/test/SqlFunctionsTest.java

```diff
@@ -810,6 +826,674 @@ private void thereAndBack(byte[] bytes) {
     assertThat(SqlFunctions.multisetUnionDistinct(z, addc),
         is(Arrays.asList("a", "c", "d")));
   }
+
+  @Test public void testJsonValueExpression() {
+    assertThat(SqlFunctions.jsonValueExpression("{}"), is(Collections.emptyMap()));
+  }
+
+  @Test public void testJsonStructuredValueExpression() {
+    assertThat(SqlFunctions.jsonStructuredValueExpression("bar"),
+        is("bar"));
+    assertThat(SqlFunctions.jsonStructuredValueExpression(100),
+        is(100));
+  }
+
+  private void assertJsonApiCommonSyntax(Object input, String pathSpec,
+                                         SqlFunctions.PathContext expected) {
+    assertThat(SqlFunctions.jsonApiCommonSyntax(input, pathSpec),
+        new BaseMatcher<SqlFunctions.PathContext>() {
+          @Override public boolean matches(Object item) {
+            if (!(item instanceof SqlFunctions.PathContext)) {
+              return false;
+            }
+            SqlFunctions.PathContext context = (SqlFunctions.PathContext) item;
+            if (Objects.equals(context.mode, expected.mode)
+                && Objects.equals(context.pathReturned, expected.pathReturned)) {
+              if (context.exc == null && expected.exc == null) {
+                return true;
+              }
+              return context.exc != null && expected.exc != null
+                  && Objects.equals(context.exc.getClass(), expected.exc.getClass())
+                  && Objects.equals(context.exc.getMessage(), expected.exc.getMessage());
+            }
+            return false;
+          }
+
+          @Override public void describeTo(Description description) {
+            description.appendText("is ").appendText(expected.toString());
+          }
+        });
+  }
+
+  @Test public void testJsonApiCommonSyntax() {
+    assertJsonApiCommonSyntax(ImmutableMap.of("foo", "bar"), "lax $.foo",
+        SqlFunctions.PathContext.withReturned(SqlFunctions.PathMode.LAX, "bar"));
+    assertJsonApiCommonSyntax(ImmutableMap.of("foo", "bar"), "strict $.foo",
+        SqlFunctions.PathContext.withReturned(SqlFunctions.PathMode.STRICT, "bar"));
+    assertJsonApiCommonSyntax(ImmutableMap.of("foo", "bar"), "lax $.foo1",
+        SqlFunctions.PathContext.withReturned(SqlFunctions.PathMode.LAX, null));
+    assertJsonApiCommonSyntax(ImmutableMap.of("foo", "bar"), "strict $.foo1",
+        SqlFunctions.PathContext.withStrictException(
+            new PathNotFoundException("No results for path: $['foo1']")));
+    assertJsonApiCommonSyntax(ImmutableMap.of("foo", 100), "lax $.foo",
+        SqlFunctions.PathContext.withReturned(SqlFunctions.PathMode.LAX, 100));
+  }
+
+  @Test public void testJsonExists() {
```

Comment:
Would you please ensure the tests print human-understandable messages in case of failures?
Note: you might find `@Parameterized` tests convenient to enumerate test cases.